src/player.cpp

```diff
@@ -10531,12 +10531,8 @@ hint_rating player::rate_action_disassemble( const item &it )
 hint_rating player::rate_action_use( const item &it ) const
 {
     if( it.is_tool() ) {
-        const auto tool = dynamic_cast<const it_tool*>(it.type);
-        if (tool->charges_per_use != 0 && it.charges < tool->charges_per_use) {
-            return HINT_IFFY;
-        } else {
-            return HINT_GOOD;
-        }
+        return it.ammo_remaining() < it.ammo_required() ? HINT_IFFY : HINT_GOOD;
```

Comment:
This idiom occurs so frequently it's tempting to provide `bool ammo_sufficient() const`?